pkg/rbac/middleware.go

```diff
@@ -31,45 +54,81 @@ func Middleware() func(http.Handler) http.Handler {
 			// Retrieving and verifying the token
 			if clientToken == "" {
 				klog.V(4).Info("Request didn't have a valid authentication token.")
-				http.Error(w, "{\"message\":\"Request didn't have a valid authentication token.\"}", http.StatusUnauthorized)
-				return
-			}
-			authenticated, err := verifyToken(clientToken, r.Context())
-			if err != nil {
-				klog.Warning("Unexpected error while authenticating the request token.", err)
-				http.Error(w, "{\"message\":\"Unexpected error while authenticating the request token.\"}", http.StatusInternalServerError)
+				http.Error(w, "{\"message\":\"Request didn't have a valid authentication token.\"}",
+					http.StatusUnauthorized)
 				return
 			}
-			if !authenticated {
-				klog.V(4).Info("Rejecting request: Invalid token.")
-				http.Error(w, "{\"message\":\"Invalid token\"}", http.StatusForbidden)
-				return
+			//check if token review doesn't already exists or if token has expired
+			// (difference between the creation time and exp time is more than 1 min.)
+			if !utr.DoesTokenExist(clientToken) || utr.ValidatedTokens[clientToken].Sub(utr.expiresAt) > duration_Minute {
+				utr.Remove(clientToken)
```

Comment:
Don't delete the user data here.
Re-validate the token first. If the token is still valid we can continue using the cached data.

Reply:
It won't make a big difference now, but remember that we will be caching more things for the user.

Reply:
okay, then in that case we don't even need to delete since the validation function will send error if it's not valid - at that point we delete.